tests/unit/migrations/add_maintenance_window_start_spec.sh

```diff
@@ -0,0 +1,50 @@
+Describe 'add-maintenance-window-start migration'
+	BeforeEach 'setup'
+	AfterEach 'teardown'
+	setup()
+	{
+		SPEC_TMPDIR="$(mktemp --directory --tmpdir nextcloud-snap-tests.XXXXXXXXXX)"
+		MIGRATION="src/migrations/migrations/28.0.0snap1/1_add-maintenance-window-start.sh"
+
+		# Create nextcloud config for the migration to tweak
+		mkdir -p "$SPEC_TMPDIR/nextcloud/config"
+		CONFIG_FILE="$SPEC_TMPDIR/nextcloud/config/config.php"
+		touch "$CONFIG_FILE"
+
+		# Make the migration look in the right place
+		export SNAP_DATA="$SPEC_TMPDIR"
+	}
+
+	teardown()
+	{
+		rm -rf "$SPEC_TMPDIR"
+	}
+
+	It 'handles empty config file'
+		When call "$MIGRATION"
+		The status should be failure
+	End
+
+	It 'handles basic config file'
```

Comment:
I'd like to see three more test cases, and I think you'll see this still needs a smidge more work:

1. `It 'handles one line config file'`, using the following config:
```
?php
$CONFIG = array();
```

2. `It 'handles more complex config file'`, using the following config:
```
?php
$CONFIG = array(
'memcache.locking' => '\OC\Memcache\Redis',
'memcache.local' => '\OC\Memcache\Redis',
);
```

3. `It 'handles broken config file'`, using the following config (it should barf on this one instead of mangling it further):
```
?php
$CONFIG = array(
```

Reply:
I think we might get stuck on this in that it almost requires a PHP interpreter written in bash. Maybe we should use nextcloud-fixer instead, where we can use OCC to do this.

Reply:
You mean [here](https://github.com/nextcloud-snap/nextcloud-snap/tree/master/src/nextcloud/fixes/existing-install) and [here](https://github.com/nextcloud-snap/nextcloud-snap/tree/master/src/nextcloud/fixes/fresh-install)?
On fresh install it just can add the config. On existing it would need to check if it already exists and only add if not exists.

Reply:
Exactly. For the fresh install, though, we can probably just put it straight [in here](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/src/nextcloud/config/config.php).

Reply:
Changed it to use migrations. I think it's fine as step 7 here without having maintenance mode enabled. I checked the used occ commands but there isn't a more easy way to add a config value only if the config value is empty.

But two more questions here:
1. What about tests? Didn't see tests for the migration steps (maybe I'm just blind).
2. Should I also revert the change from [tests/unit/spec_helper.sh](https://github.com/nextcloud-snap/nextcloud-snap/pull/2659/files#diff-93c02d4295748a9832e89fbc425d8fb0be7311652ca219a680970ba947adedcdL15) or do you want to keep them?

Reply:
> What about tests?

I would say they are not needed anymore, as we are using the official tool to update the config and not a bash script.